cmd/agent/main.go

```diff
@@ -181,6 +181,15 @@ func registerWatchers(ctx context.Context) error {
 		watchersEnabled = append(watchersEnabled, w)
 	}
 
+	if global.AgentConf.Discovery.Disabled {
```

Comment:
I think it might make sense to differentiate between 'configuration' and 'supported'/'implemented' functionality here.

Reply:
`global.AgentConf` holds the effective configuration that the agent will use after all config variants have been sourced. But I modified discovery code to source supported functionality from the protocol's implementation here https://github.com/Metrika-Inc/agent/pull/94/files#diff-5c94f018f6eb4ca8ecf273cb057a66b0b6d357b5c1473de0c504ad380c9607acR61